src/microbe_stage/editor/PatchMapNode.cs

```diff
@@ -140,5 +189,14 @@ private void UpdateIcon()
             return;
 
         iconRect.Texture = PatchIcon;
+        iconDirty = false;
+    }
+
+    private void UpdateGrayscale()
+    {
+        if (iconRect != null)
+            iconRect.Material = Enabled ? null : MonochromeShader;
+
+        grayscaleDirty = false;
```

Comment:
Should the dirty flag be cleared when the material could not be actually applied yet?